cmd/manager/main_test.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/version"
+)
+
+const (
+	nonReleaseVersion = "1.8.0-30-g81233c6-dirty"
+	releaseVersion    = "1.9.0-certified"
+)
+
+func TestDeletionProtectionEnable(t *testing.T) {
+	testCases := []struct {
```

Comment:
Suggestion: when we have that many options to setup a test, let's avoid table testing, IMO it's very hard to see what's actually being tested.